@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -501,56 +522,84 @@ function applyTechniquePropertyToMaterial(
 ) {
     const m = material as any;
     if (m[prop] instanceof THREE.Color) {
-        if (typeof value === "string") {
-            value = parseStringEncodedColor(value);
-            if (value === undefined) {
-                throw new Error(`Unsupported color format: '${value}'`);
-            }
+        applyTechniqueColorToMaterial(technique, material, prop, value, level);
+    } else {
+        m[prop] = value;
+    }
+}
+
+/**
+ * Apply technique color to material taking special care with transparent (RGBA) colors.
+ *
+ * @note Transparent colors should be processed as the very last technique attributes,
+ * since their effect on [[THREE.Material]] properties like __opacity__ and __transparent__
+ * could be overridden by other technique params.
+ *
+ * @param technique an technique the applied color comes from
+ * @param material the material to which color is applied
+ * @param prop technique property (color) name
+ * @param value color value
+ * @param level optional, tile zoom level for zoom-level dependent properties are evaluated.
+ */
+function applyTechniqueColorToMaterial(
```

Comment:
This is almost same as `applyColorToMaterial` (DecodedTileHelpers) added in #1104, https://github.com/heremaps/harp.gl/pull/1104/files#diff-4414054ad69d5d73485b5a4b589904e7R1573

Shouldn't we merge these two ?

Reply:
And this should be named `applyMainTechniqueColorToMaterial` because this particular color should have some "afjective" and main is good :) (IMO)

Reply:
I guess you meant TileGeometryCreator.applyColorToMaterial(..), that's the fact that both methods are almost the same, the main three differences are:
- In TileGeometryCreator zoom level (`level` param) is always defined - don't ask me why it's not the same in DecodedTileHelpers - I am not the owner of original code/interfaces.
- In TileGeometryCreator all technique properties are always interpolated, which is not true for DecodedTileHelpers (see above for comments :) )
- In DecodedTileHelpers we could potentially use pre-calculated Material.opacity value (without call to getPropertyValue) assuming that transparent color transfer to material goes as the last,
- DecodedTileHelpers triggers notifying change (again legacy code) on material:
`m[prop] = m[prop];`
which is not done in TileGeometryCreator.
Summarizing, if we could use one general method for both, that would need to be DecodedTileHelpers.applyColorToMaterial(...) with all the consequences - lack of simplification and with all `level` checks and so...
I am also voting for this, but need some hints if it would be still correct and desired...